packages/emotion/native/Css.ml

```diff
@@ -228,31 +228,33 @@ let pp_rules hash rules =
     |> List.flatten
     |> List.filter_map render_declaration
     |> String.concat " "
-    |> fun all -> Printf.sprintf ".%s { %s }" hash all
+    |> fun all -> Printf.sprintf ".%s { %s }" className all
   in
   let selectors =
     list_of_rules
-    |> List.filter_map (render_selectors hash)
+    |> List.filter_map (render_selectors className)
     |> String.concat " "
   in
   Printf.sprintf "%s %s" declarations selectors
 
 type declarations =
+  | Global of rule list
   | Classnames of rule list
   | Keyframes of (int * rule list) list
 
 module Stylesheet = struct
   module Hashes = Set.Make (String)
 
   type 'a t = {
-    mutable rules : (string * declarations) list;
+    (* rules: (prefix * hash * rules) *)
+    mutable rules : (string * string * declarations) list;
```

Comment:
maybe a record here would be needed now, wdyt?

Reply:
Due to your comment below (https://github.com/davesnx/styled-ppx/pull/467#discussion_r1580874013)

I was think to keep as before:
```suggestion
    mutable rules : (string * declarations) list;
```

And have the declarations like this:
```ocaml
type declarations =
  | Global of rule list
  | Classnames of (string * rule list)
  | Keyframes of (string * (int * rule list) list)
```

Maybe declarations Classnames and Keyframes could be a record for better documentation.

```ocaml
type declarations =
  | Global of rule list
  | Classnames of {
      className : string;
      styles : rule list;
    }
  | Keyframes of {
      animationName : string;
      keyframes : (int * rule list) list;
    }

```

WDYT?